wcomponents-core/src/main/java/com/github/bordertech/wcomponents/AbstractWComponent.java

```diff
@@ -1278,8 +1279,7 @@ public boolean isDefaultState() {
 	 * @return the number of child components currently contained within this component.
 	 */
 	int getChildCount() {
-		ComponentModel model = getComponentModel();
-		return (model.getChildren() == null ? 0 : model.getChildren().size());
+		return getChildren().size();

```

Comment:
As above...again it would be better to just get the size off the model without creating another list.

If you can go through the PR and apply this principle, that would be great. "@" me when you have made the changes and I will take another look.

Thanks again for the PR as it will be a good API change for application developers.